### PR TITLE
[build] Setup to build ATen from root CMake file

### PR DESCRIPTION
**Pull request by orionr**
* Cleanup aten/ cmake files in prep for including it from the root-level cmake

@ezyang and @Yangqing please review. Thanks.

### CODE REVIEW

**Comment by Yangqing** [comment]
FindCuDNN.cmake might not be needed because Caffe2's cuda helper actually addressed it (with modern cudnn targets).

File: aten/CMakeLists.txt [hunk 1]
```diff
@@ -1,12 +1,20 @@
...
+    return()
+  endif()
+else()
+  cmake_minimum_required(VERSION 3.0 FATAL_ERROR)
+  project(ATen CXX C)
+endif()
+
+# Find modules
+list(APPEND CMAKE_MODULE_PATH
+  ${CMAKE_CURRENT_SOURCE_DIR}/../cmake/Modules
```
Comment (Yangqing) [inline aten/CMakeLists.txt]: I think this line combined with the fact that PR #6699 nuked FindMKL.cmake in /cmake causes the MKL build to fail. I have an ongoing PR #7170 to address this problem.

**Comment by Yangqing** [comment]
Also see #7170  cc @ezyang @zdevito

File: aten/CMakeLists.txt [hunk 2]
```diff
@@ -1,12 +1,19 @@
-cmake_minimum_required(VERSION 3.0)
-set(CMAKE_MODULE_PATH
-  ${CMAKE_CURRENT_SOURCE_DIR}/cmake
-  ${CMAKE_CURRENT_SOURCE_DIR}/../cmake/Modules_CUDA_fix
+if (CAFFE2_CMAKE_BUILDING_WITH_MAIN_REPO)
```
Comment (ezyang) [inline aten/CMakeLists.txt]: What is the planned terminal name for this variable?

Comment (Yangqing) [reply aten/CMakeLists.txt]: Yep, this is used in modules. In the future we might want to call it something like "BUILDING_AS_STANDALONE_PROJECT" but it really doesn't matter.

**Comment by orionr** [comment]
Except for FindMKL.cmake the other Find*.cmake files were luckily bit-for-bit identical.

**Review by bddppq — APPROVED** [review]

**Review by ezyang — APPROVED** [review]

**Pull request merged**